File: TP 3 - C/ejecutar_comando.c
#include "ejecutar_comando.h"
#include "listar.h"
#include "contactar.h"
#include "modificar.h"
#include "agregar.h"
#include "comandos.h"
#include <stdlib.h>

int activar_comando(char* argv[], FILE* superheroes, char ruta[MAX_RUTA]){

    int codigo = NO_HAY_ERROR;
    
    if(strcmp(argv[1],LISTAR) == 0){
        codigo = listar_superheroes(superheroes);
    }else if(strcmp(argv[1],AGREGAR) == 0){
        codigo = agregar_superheroes(superheroes, atoi(argv[2]), argv[3], atoi(argv[4]), argv[5], ruta);
    }else if(strcmp(argv[1],CONTACTAR) == 0){
        codigo = contactar_superheroes(superheroes, atoi(argv[2]), ruta);
    }else if(strcmp(argv[1],MODIFICAR) == 0){
        codigo = modificar_superheroes(superheroes, atoi(argv[2]), atoi(argv[3]), argv[4], ruta);
    }

    return codigo;
}

File: TP3 - PYTHON/ejecutar_comando.py
import codigo_error as code
import comandos as cmd
from listar import listar_superheroes
from modificar import modificar_superheroes
from agregar import agregar_superheroes
from contactar import contactar_superheroes

#PRE: - 
#POS: Retorna 0 si no hubo error. Sino el código correspondiente de error.
def activar_comando(argv, superheroes, ruta):

    codigo = code.NO_HAY_ERROR

    if(argv[1] == cmd.LISTAR):
        codigo = listar_superheroes(superheroes)
    elif(argv[1] == "contactar"):
        codigo = contactar_superheroes(superheroes, argv[2])
    elif(argv[1] == "modificar"):
        codigo = modificar_superheroes(superheroes, argv[2], argv[3], argv[4])
    elif(argv[1] == "agregar"):
        codigo = agregar_superheroes(superheroes, argv[2], argv[3], argv[4], argv[5])

    return codigo